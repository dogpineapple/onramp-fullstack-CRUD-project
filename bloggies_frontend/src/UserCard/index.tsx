import React from "react";
import { Card } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import { User } from "../custom";
import { changeToURLFriendly } from "../helpers";
import moment from "moment";
import "./UserCard.css";

interface IProp {
  user: User
}

function UserCard({ user }: IProp) {
  return (
    <Card className="UserCard text-left">
      <Card.Body>
        <Card.Text className="BlogCard-body">
          <NavLink to={`/users/${user.id}/${changeToURLFriendly(user.display_name)}/favorites`}>
            {user.display_name}
          </NavLink>
        </Card.Text>
        <Card.Subtitle>Joined {moment(user.join_date).fromNow()}</Card.Subtitle>
      </Card.Body>
    </Card>
  );
};

export default UserCard;