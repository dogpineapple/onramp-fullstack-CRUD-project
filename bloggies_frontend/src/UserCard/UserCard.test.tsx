import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserCard from './index';
import { MOCK_USER } from '../jest.mock';

Enzyme.configure({ adapter: new Adapter() });

describe('UserCard', () => {
  it('should show user information', () => {

    const wrapper = shallow(<UserCard user={MOCK_USER}/>);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
