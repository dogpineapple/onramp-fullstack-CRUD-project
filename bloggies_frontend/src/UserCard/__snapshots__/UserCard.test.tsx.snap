// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should match snapshot 1`] = `
<Card
  body={false}
  className="UserCard text-left"
>
  <CardBody>
    <CardText
      className="BlogCard-body"
    >
      <NavLink
        to="/users/1/test-username/favorites"
      >
        Test Username
      </NavLink>
    </CardText>
    <CardSubtitle>
      Joined 
      a year ago
    </CardSubtitle>
  </CardBody>
</Card>
`;
